ddtrace/llmobs/_llmobs.py

```diff
@@ -542,6 +551,16 @@ def annotate(
             else:
                 cls._tag_text_io(span, input_value=input_data, output_value=output_data)
 
+    @staticmethod
+    def _tag_prompt(span, prompt: dict) -> None:
+        """Tags a given LLMObs span with a prompt"""
+        try:
+            validated_prompt = validate_prompt(prompt)
+            span.set_tag_str(INPUT_PROMPT, json.dumps(validated_prompt))
+        except TypeError:
+            log.warning("Failed to validate prompt with error: ", exc_info=True)
+            return
```

Comment:
```suggestion
        try:
            validated_prompt = validate_prompt(prompt)
        except TypeError:
            log.warning("Failed to validate prompt with error: ", exc_info=True)
            return
        span.set_tag_str(INPUT_PROMPT, json.dumps(validated_prompt, default=_unserializable_default_repr))
```
Let's make sure we safely json dump instead of erroring for unserializable objects, the type checking / validating can be a separate try/catch